engines/groovie/music.cpp

```diff
@@ -793,8 +793,15 @@ bool MusicPlayerTlc::load(uint32 fileref, bool loop) {
 	unload();
 	_file = new Common::File();
 
-	// Create the audio stream from fileref
 	Common::String filename = getFilename(fileref);
+
+	// Apple platforms use m4a files instead of mpg
+	if (_vm->getPlatform() == Common::kPlatformMacintosh || _vm->getPlatform() == Common::kPlatformIOS)
+		filename += ".m4a";
```

Comment:
Turns out that if you load a `.m4a` with `Audio::makeMP3Stream()` below, then the first note keeps repeating. Instead, `Audio::makeQuickTimeStream()` does load the music properly. So you'll need to switch which way you are loading the file based on which extension we have (or as we speculated in Discord, see if there's a way ScummVM has to automatically pick what type of stream to use).